Telegraf crashes with the followinfg error: panic: runtime error: invalid memory address or nil pointer dereference

I was running telegraf version 1.0.0 on CentOS release 6.5 when it crashed. Here are the last few lines from /var/log/telegraf/telegraf.log:

2016/11/18 10:18:30 Output [influxdb] wrote batch of 1000 metrics in 23.381351ms
2016/11/18 10:18:30 Output [influxdb] wrote batch of 1000 metrics in 23.187129ms
2016/11/18 10:18:30 Output [influxdb] wrote batch of 1000 metrics in 22.511862ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x40 pc=0x590035]

goroutine 5344346 [running]:
panic(0x12cefe0, 0xc82000e0c0)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/filestat.(*FileStat).Gather(0xc820134f30, 0x7fb15bb3b710, 0xc823395770, 0x0, 0x0)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/filestat/filestat.go:82 +0x675
github.com/influxdata/telegraf/agent.gatherWithTimeout.func1(0xc8202ace40, 0xc820134f90, 0xc823395770)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:163 +0x73
created by github.com/influxdata/telegraf/agent.gatherWithTimeout
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:164 +0xe0

Is this fixed in version 1.1.1 ? Or are there any workarounds?

Let me know if you require any other info.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@
 - [#2146](https://github.com/influxdata/telegraf/issues/2146): Fix potential panic in aggregator plugin metric maker.
 - [#1843](https://github.com/influxdata/telegraf/pull/1843) & [#1668](https://github.com/influxdata/telegraf/issues/1668): Add optional ability to define PID as a tag.
 - [#1730](https://github.com/influxdata/telegraf/issues/1730): Fix win_perf_counters not gathering non-English counters.
+- [#2061](https://github.com/influxdata/telegraf/issues/2061): Fix panic when file stat info cannot be collected due to permissions or other issue(s).
 
 ## v1.1.2 [2016-12-12]
 

--- a/plugins/inputs/filestat/filestat.go
+++ b/plugins/inputs/filestat/filestat.go
@@ -4,6 +4,7 @@ import (
 	"crypto/md5"
 	"fmt"
 	"io"
+	"log"
 	"os"
 
 	"github.com/influxdata/telegraf"
@@ -78,8 +79,14 @@ func (f *FileStat) Gather(acc telegraf.Accumulator) error {
 				"file": fileName,
 			}
 			fields := map[string]interface{}{
-				"exists":     int64(1),
-				"size_bytes": fileInfo.Size(),
+				"exists": int64(1),
+			}
+
+			if fileInfo == nil {
+				log.Printf("E! Unable to get info for file [%s], possible permissions issue",
+					fileName)
+			} else {
+				fields["size_bytes"] = fileInfo.Size()
 			}
 
 			if f.Md5 {

